app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Any:
+    stored_cache = {}
+
+    def wrapper(*args: (int, float, str), **kwargs: (tuple, bool)) -> Any:
+        if args not in stored_cache:
+            result = func(*args, **kwargs)
+            stored_cache[args] = result
```

Comment:
Write this inline

Reply:
And then move return from if-else statement

Reply:
Didn't really get it. Do I need to write lines 9 and 10 in one line? And about return also didn't understand, sorry